vmf/scripts/mods/vmf/modules/vmf_mod_manager.lua

```diff
@@ -115,6 +115,7 @@ function vmf.initialize_mod_data(mod, mod_data)
   vmf.set_internal_data(mod, "is_togglable",    mod_data.is_togglable or mod_data.is_mutator)
   vmf.set_internal_data(mod, "is_mutator",      mod_data.is_mutator)
   vmf.set_internal_data(mod, "allow_rehooking", mod_data.allow_rehooking)
+  vmf.set_internal_data(mod, "workshop_id",     mod_data.workshop_id)
```

Comment:
I think we should try to automatically fill `workshop_id` because:

+ It's more convenient for the user, and less error prone.
+ Makes it easier to fork mods of others.
+ Eliminates the need to provide two IDs for mods with beta/release versions.


The follow patch (made with help from @Azumgi) should be able to determine the `workshop_id` for most mods, while still allowing it to be overriden for advanced use cases. _Warning: untested code & written from a browser._

```suggestion
local workshop_id = mod_data.workshop_id
local mod_manager = Manager.mods
local current_mod_load_index = mod_manager._mod_load_index
if not workshop_id and current_mod_load_index then
  current_mod_load_index 
  workshop_id = mod_manager._mods[current_mod_load_index].id
end
vmf.set_internal_data(mod, "workshop_id", workshop_id)
```
(source: https://discordapp.com/channels/404978161726783489/454308427766693888/459282800793747457)

Reply:
As I told before `workshop_id` should not be passed in `mod_data` at all. There is just no point in that. Just straight up save ID from vanilla mod manager.

As for the piece of code in discord, I didn't test it. Even though, it should work.

Also, why mod_handle is not saved as internal data as well? It should be saved as internal data if there's no reason not to.